streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollectorImpl.java

```diff
@@ -107,11 +107,11 @@ public void onCompletion(final RecordMetadata metadata, final Exception exceptio
                         } else {
                             if (sendException == null) {
                                 sendException = exception;
-                                if (sendException instanceof ProducerFencedException) {
-                                    log.error("{} Error sending record to topic {}. No more offsets will be recorded for this task and it will be closed as it is a zombie.", logPrefix, topic, exception);
-                                } else {
-                                    log.error("{} Error sending record to topic {}. No more offsets will be recorded for this task and the exception will eventually be thrown", logPrefix, topic, exception);
-                                }
+                                log.error("{} Error sending record with key {} to topic {}, partition {}. " +
```

Comment:
Because at a higher level, we catch `ProducerFenced` exception. If we wrap it, it would bubble throw and kill the thread -- but there is no reason to kill the thread. We just close the task as zombie.

From my understanding, throwing a `StreamsException` at this point indicate a fatal error -- but ProducerFenced is not fatal, thus we don't wrap it.